### PR TITLE
libuutil: deobfuscate internal pointers

### PR DESCRIPTION
**Pull request by brooksdavis**
### Motivation and Context
libuutil's uu_avl and uu_list data structures use a pointer obfuscation which is incompatible with CHERI pointers (byte swapping the address takes the pointer too far out of range) so I needed to disable it on CheriBSD. Simply removing the obfuscation is cleaner and modern leak sanitizers should be able to handle this case.

### Description
uu_avl and uu_list stored internal next/prev pointers and parent pointers (unused) obfuscated (byte swapped) to hide them from a long forgotten leak checker (No one at the 2022 OpenZFS developers meeting could recall the history.)  This would break on CHERI systems and adds no obvious value.  Rename the members, use proper types rather than uintptr_t, and elimate the related macros.

I've checked the library ABI box below as the .abi files need to be updated, but the change is and API change on internal struct members and the structs don't change size.

### How Has This Been Tested?
sanity tests on FreeBSD.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

Requires-builders: test

### CODE REVIEW

**Review by ryao — APPROVED** [review]

**Comment by brooksdavis** [comment]
Hmm, `Requires-builders: all` didn't seem to tigger builds. I've tried switching to `test`, but I'm not sure if the builders will trigger based on that edit. I believe I need a build that complains about .abi mismatches so I can update the files.

**Comment by ryao** [comment]
@brooksdavis Since you are a new contributor, the builders might require approval from someone with write access to the repository before they will run. That is limited to a small group for security purposes. @behlendorf should be able to tell them to run.

**Comment by behlendorf** [comment]
@brooksdavis can you try force updating this branch again.  For some reason the GitHub webhook seems to have failed to run.

**Pull request merged**
